### PR TITLE
Fix MDM Agent process stop

### PR DESCRIPTION
**Pull request by joenpera**
Jira-Id: SECO-4737

### CODE REVIEW

File: modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py [hunk 1]
```diff
@@ -854,24 +841,36 @@ async def stop():
...
 
-    def signal_handler():
+
+    def signal_handler(signum, frame):
+        global mdm_agent_stopped
+
+        sig_name = signal.Signals(signum).name
+        cc.logger.debug(f"signal_handler {sig_name} ({signum}) received.")
+
+        if sig_name == "SIGUSR1":
```
Comment (pentestiing) [inline modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py:851]: Shouldn't SIGINT and SIGTERM also be checked for here? Maybe someone gets impatient and tries to stop the MDM agent twice in a row :thinking:

File: modules/sc-mesh-secure-deployment/src/nats/src/cbma_adaptation.py [hunk 2]
```diff
@@ -540,7 +558,7 @@ def __setup_radios(self) -> bool:
 
         return True
 
-    def setup_cbma(self) -> None:
+    def setup_cbma(self) -> bool:
```
Comment (pentestiing) [inline modules/sc-mesh-secure-deployment/src/nats/src/cbma_adaptation.py:561]: It doesn't look like this function can return `False` :heavy_check_mark:

File: modules/sc-mesh-secure-deployment/src/nats/src/comms_if_monitor.py [hunk 3]
```diff
@@ -94,6 +96,24 @@ def monitor_interfaces(self):
...
+    def __create_dummy_interface(self, interface_name):
+        """
+        Create a dummy interface.
+
+        Arguments:
+            interface_name: str -- Name of the interface to be created.
+        """
+        subprocess.run(['ip', 'link', 'add', 'dev', interface_name, 'type', 'dummy'],
+                           check=True)
+    def __delete_dummy_interface(self, interface_name):
```
Comment (pentestiing) [inline modules/sc-mesh-secure-deployment/src/nats/src/comms_if_monitor.py:108]: Newline missing before :baby_bottle:

**Review by pentestiing — APPROVED** [review]
Supremely superb! :superhero:

File: modules/sc-mesh-secure-deployment/src/nats/src/cbma_adaptation.py (hunk 2 above)
Comment (joenpera) [reply modules/sc-mesh-secure-deployment/src/nats/src/cbma_adaptation.py:561]: yes, but there is "return true" at the end and code editors are complaining "-> None vs return true".   Fail case handling needs to added.

**Review by saauvine — APPROVED** [review]

**Pull request merged**